src/components/time-picker/time-picker.js

```diff
@@ -0,0 +1,97 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+
+class TimePicker extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Time picker UI.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as a time picker UI.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this._initInputEvents();
+    this._setMinMaxValues();
+    this.element.militaryTime = false;
+  }
+
+  setMilitaryTime = () => {
+    this.element.militaryTime = true;
+    this._setMinMaxValues();
+  }
+
+  _setMinMaxValues = () => {
+    const hourInput = this.element.querySelector(this.options.classHoursInput);
+    if (this.element.militaryTime) {
+      hourInput.max = 24;
```

Comment:
12/24 hours preferences depends on locales, so I'd suggest letting time parser (like one in Flatpickr) validate the time.